src/TemplateBuilder/RootWizard.cs

```diff
@@ -1,36 +1,268 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 using EnvDTE;
+using EnvDTE100;
+using EnvDTE80;
 using Microsoft.VisualStudio.TemplateWizard;
 
-namespace TemplateBuilder {
-    public class RootWizard : IWizard {
+namespace TemplateBuilder
+{
+    public class RootWizard : IWizard
+    {
+        // Core Solution object
+        private Solution4 _solution;
+
         // Use to communicate $saferootprojectname$ to ChildWizard     
         public static Dictionary<string, string> GlobalDictionary = new Dictionary<string, string>();
 
         // Add global replacement parameters     
-        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams) {
+        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams)
+        {
             // Place "$saferootprojectname$ in the global dictionary. 
             // Copy from $safeprojectname$ passed in my root vstemplate         
+            GlobalDictionary["$destinationdirectory$"] = replacementsDictionary["$destinationdirectory$"];
             GlobalDictionary["$saferootprojectname$"] = replacementsDictionary["$safeprojectname$"];
+
+            if (runKind == WizardRunKind.AsMultiProject)
+            {
+                var dte2 = automationObject as DTE2;
+                if (dte2 != null) _solution = (Solution4)dte2.Solution;
+            }
         }
 
-        public bool ShouldAddProjectItem(string filePath) {
+        public bool ShouldAddProjectItem(string filePath)
+        {
             return true;
         }
 
-        public void RunFinished() {
+        public void RunFinished()
+        {
+            if (_solution == null)
+            {
+                throw new Exception("No solution found!");
+            }
+
+            //Get all projects in solution
+            var projects = GetProjects();
```

Comment:
@Reagann per your last comment, it looks like you're using you're custom ```GetProjects()``` method to get all the projects in the solution. Before you do anything with the multiple project template, could you not use this same method to determine if there are not any already existing projects in the solution? This would provide the separation you are looking for.